packages/roosterjs-content-model-core/lib/coreApi/setContentModel/setContentModel.ts

```diff
@@ -15,6 +18,28 @@ import type { SetContentModel } from 'roosterjs-content-model-types';
  */
 export const setContentModel: SetContentModel = (core, model, option, onNodeCreated) => {
     const editorContext = core.api.createEditorContext(core, true /*saveIndex*/);
+
+    if (option?.shouldMaintainSelection) {
```

Comment:
This need to move down after line 57, and use the selection range from the return value, but not rely on CSS. otherwise what if there are multiple selected elements